src/Query/ResultSetMapping.php

```diff
@@ -159,6 +159,13 @@ class ResultSetMapping
      */
     public array $newObjectMappings = [];
 
+    /**
+     * Maps object Ids in the result set to classnames.
+     *
+     * @phpstan-var array<string|int, string>
```

Comment:
If it's a `class-string`, let's say so?

```suggestion
     * @phpstan-var array<string|int, class-string>
```